source/admin_manual/list_of_events.rst

```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
-| dest_port           | TCP connection's destination port.                   |
-+---------------------+------------------------------------------------------+
-| dest_host           | TCP connection's destination hostname, if known.     |
-+---------------------+------------------------------------------------------+
-| socket_path         | UNIX socket connection's path                        |
-+---------------------+------------------------------------------------------+
-| remote_pid          | Remote UNIX socket server's process ID.              |
-+---------------------+------------------------------------------------------+
-| remote_uid          | Remote UNIX socket server's system user ID.          |
-+---------------------+------------------------------------------------------+
-
-client_connection_connected
----------------------------
-
-Emitted when a server accepts an incoming client connection.
-
-
-client_connection_disconnected
-------------------------------
-
-Emitted when a client connection is terminated.
-
-+---------------------+------------------------------------------------------+
-| Field               | Description                                          |
-+=====================+======================================================+
-| bytes_in            | Amount of data read, in bytes                        |
-+---------------------+------------------------------------------------------+
-| bytes_out           | Amount of data written, in bytes                     |
-+---------------------+------------------------------------------------------+
-| reason              | Disconnection reason                                 |
-+---------------------+------------------------------------------------------+
-
-
-server_connection_connected
----------------------------
-
-Emitted when an outgoing server connection was either successfully established
-or failed. Currently it is not possible to know which one happened.
-
-server_connection_disconnected
-------------------------------
-
-Emitted when a server connection is terminated.
-
-+---------------------+------------------------------------------------------+
-| Field               | Description                                          |
-+=====================+======================================================+
-| bytes_in            | Amount of data read, in bytes                        |
-+---------------------+------------------------------------------------------+
-| bytes_out           | Amount of data written, in bytes                     |
-+---------------------+------------------------------------------------------+
-| reason              | Disconnection reason                                 |
-+---------------------+------------------------------------------------------+
 
 FS
 ==
 
-.. _event_fs:
+.. dovecot_core:event:: fs
 
-fs
---
+   May be inherited from various different parents (e.g. "Mail User" event)
```

Comment:
Why not link to "mail user" event?

Reply:
There is no "mail user" event.  There's a inherited group of attribute/fields called "mail_user", but it can't be linked to separately.  And regardless, should we really be linking just mail user here?  If we were to add links, we would need to somehow list all the things that could be inherited from (or else people are just going to assume that mail user is the only thing that matter here).

And this leads to my biggest current issue with events - these polymorphic events are extremely confusing.  i.e. an event can have different fields based on what triggered it.  This is very troublesome for anybody who is trying to create input filters or parsers on a system that can't handle different fields for the same event.

Reply:
>  There's a inherited group of attribute/fields called "mail_user", but it can't be linked to separately

Why can't we link to it?

>  And regardless, should we really be linking just mail user here? If we were to add links, we would need to somehow list all the things that could be inherited from (or else people are just going to assume that mail user is the only thing that matter here).

Could be a problem, yes..

> And this leads to my biggest current issue with events - these polymorphic events are extremely confusing. i.e. an event can have different fields based on what triggered it. This is very troublesome for anybody who is trying to create input filters or parsers on a system that can't handle different fields for the same event.

Well, that can't really be helped. We can't either create tons of different events depending on what the code path was to create them. I haven't heard of this being an issue though. Is it a real problem somewhere? I'd think that even in that case it should be possible to configure, just not configure `event=foo` but rather checking more fields like `event=foo AND NOT box=""`